src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int SALARY_INDEX = 3;
+    private static final int NUMBER_OF_INDEXES = 4;
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        LocalDate dateFromFormatted = getDate(dateFrom);
+        LocalDate dateToFormatted = getDate(dateTo);
+        LocalDate dateInOneRowFromData;
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            int income = 0;
+            for (int j = 0; j < data.length; j++) {
+                dateInOneRowFromData = getDate(splitData(data[j], DATE_INDEX));
+                if (names[i].equals(splitData(data[j], NAME_INDEX))
```

Comment:
```suggestion
            for (int j = 0; j < data.length; j++) {
                String[] record = data[j].split(":");
                dateInOneRowFromData = getDate(record[DATE_INDEX]);
                if (names[i].equals(record[NAME_INDEX]))
```
no need to split the string multiple times, just split it once